### PR TITLE
UGENE-7335 SAFE_POINT after confirmation in "File Modification Detected" dialog

### PR DESCRIPTION
**Pull request by rasputinkirill**

### CODE REVIEW

File: src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp [hunk 1]
```diff
@@ -380,6 +386,14 @@ void PCRPrimerDesignForDNAAssemblyOPWidget::sl_extractProduct() {
     }
 }
 
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_annotationCreationTaskFinished() {
+    CreateAnnotationsTask *t = qobject_cast<CreateAnnotationsTask *>(sender());
+    if (t->getState() != Task::State_Finished || t->isCanceled() || t->hasError()) {
+        return;
+    }
+    productsTable->setAnnotationGroup(t->getResultAnnotations().at(0)->getGroup());
```
Comment (DmitriiSukhomlinov) [inline src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp:432]: What if the 0 element isn't exist? It gotta be better to add SAFE_POINT, at least

File: src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.h [hunk 2]
```diff
@@ -86,11 +86,15 @@ private slots:
      * Show the "Open file" dialog, choose a file with other PCR sequences and set path to the corresponding line edit.
      */
     void sl_loadOtherSequenceInPcr();
+    /*
+    * Associate created resultannotations with result table.
```
Comment (DmitriiSukhomlinov) [inline src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.h]: result annotations

File: src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.h [hunk 3]
```diff
@@ -86,11 +86,15 @@ private slots:
...
+    */
     void sl_onFindTaskFinished();
     /*
     * Extract result product to file.
     */
     void sl_extractProduct();
+    void sl_annotationCreationTaskFinished();
```
Comment (DmitriiSukhomlinov) [inline src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.h:96]: Add a small description

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]
1. Open "gfp.fa"
2. set "Minimum melting temperature" to 51
3. Click "Start"
4. Wait for finish
5. Click "Start" again
6. SAFE_POINT has been triggered

**Review by smlnkcv — APPROVED** [review]

**Review by DmitriiSukhomlinov — APPROVED** [review]

**Pull request merged**